main.tf

```
@@ -132,7 +132,7 @@ module "dns_replicas" {
   stage     = "${var.stage}"
   zone_id   = "${var.zone_id}"
   records   = ["${coalescelist(aws_rds_cluster.default.*.reader_endpoint, list(""))}"]
-  enabled   = "${var.enabled == "true" && length(var.zone_id) > 0 ? "true" : "false"}"
+  enabled = "${var.enabled == "true" && var.engine_mode != "serverless" ? "true" : "false"}"
```

```suggestion
  enabled = "${var.enabled == "true" && length(var.zone_id) > 0 && var.engine_mode != "serverless" ? "true" : "false"}"
```

if `zone_id` is not provided, don't provision the DNS record